auto-value-parcel/src/main/java/com/ryanharter/auto/value/parcel/AutoValueParcelExtension.java

```diff
@@ -374,13 +376,33 @@ MethodSpec generateConstructor(List<Property> properties) {
   }
 
   FieldSpec generateCreator(ProcessingEnvironment env, TypeName autoValueType,
-      List<Property> properties, TypeName type, Map<TypeMirror, FieldSpec> typeAdapters) {
+      List<Property> properties, TypeName type, Map<TypeMirror, FieldSpec> typeAdapters,
+      List<? extends javax.lang.model.element.TypeParameterElement> typeParameters) {
+
     ClassName creator = ClassName.bestGuess("android.os.Parcelable.Creator");
-    TypeName creatorOfClass = ParameterizedTypeName.get(creator, type);
+    TypeName typeWithParameters;
+    if (!typeParameters.isEmpty()) {
+      TypeName[] elements = typeParameters.stream().map(
+              param -> {
+                TypeMirror bound = param.getBounds().get(0);
+                TypeName typeName = TypeName.get(bound);
+                // TODO: support types with multiple bounds.
+                return WildcardTypeName.subtypeOf(typeName);
+              }
+      ).toArray(TypeName[]::new);
+     typeWithParameters = ParameterizedTypeName.get((ClassName) type, elements);
```

Comment:
Instead of this unchecked cast, might it make more sense to either update the `type` parameter to restrict it to a `ClassName`, or assemble the `ParameterizedTypeName` externally and pass that in as the `type` and cast iff `type is ParameterizedTypeName`?

I'm leaning towards the latter since it simplifies retains the generality of the API without hiding the `iff !typeParameters.isEmpty() then type must be ClassName` requirement.

Reply:
Turns out this value was already a `ClassName` where originally defined. I upgraded the type and now the typecast is gone. The latter option seems less rewarding now that there's no typecast, but let me know if you feel strongly about it.